org.eclipse.m2e.tests/src/org/eclipse/m2e/tests/ProjectConfigurationManagerTest.java

```diff
@@ -178,17 +179,25 @@ public void scheduled(IJobChangeEvent event) {
 
   @Test
   public void testResolutionOfArchetypeFromRepository() throws Exception {
-    useSettings("settings2.xml");
-
-    Archetype archetype = new Archetype();
-    archetype.setGroupId("org.eclipse.m2e.its");
-    archetype.setArtifactId("mngeclipse-2110");
-    archetype.setVersion("1.0");
-    archetype.setRepository(new File("repositories/testrepo").getAbsoluteFile().toURI().toASCIIString());
-    Collection<IProject> projects = createProjectsFromArchetype("mngeclipse-2110", new MavenArchetype(archetype), null);
-    assertEquals(1, projects.size());
-    IProject project = projects.iterator().next();
-    assertTrue(project.isAccessible());
+    File templateSettings = new File("settings_archetype_repo.xml").getCanonicalFile().getAbsoluteFile();
+    String interpolated = Files.readString(templateSettings.toPath()).replace("${repoUrl}",
+        new File("repositories/testrepo").getAbsoluteFile().toURI().toASCIIString());
+    File tempFile = File.createTempFile("settings", ".xml");
```

Comment:
If you'd use [Junit-4's `TemporaryFolder` rule](https://junit.org/junit4/javadoc/latest/org/junit/rules/TemporaryFolder.html) the try-finally block would not be necessary and we have less set-up/tear-down within the test-case method itself.

Reply:
Yes, but the rule will apply to all methods, I'm not sure if that worth it...